### PR TITLE
fix: return the correct tag

### PR DESCRIPTION
**Pull request by andrewrynhard**
This changes the semver filter such that:

alpha >= beta >= stable

Signed-off-by: Andrew Rynhard <andrew@rynhard.io>

### CODE REVIEW

File: pkg/channel/filter/filter.go [hunk 1]
```diff
@@ -64,22 +66,15 @@ func FilterSemver(ch string, tags []string) (target *string) {
 			if ch == channel.StableChannel {
 				continue
 			}
-			// Ensure that we are comparing the prerelease with the requested
-			// channel.
-			pre := v2.Pre[0].String()
-			if pre != ch {
-				continue
-			}
```
Comment (andrewrynhard) [inline pkg/channel/filter/filter.go:72]: We no longer require that alphas are compared with alphas and betas are compared with betas.

**Review by smira — APPROVED** [review]

**Review by rsmitty — APPROVED** [review]

**Pull request merged**